Mage/src/main/java/mage/abilities/mana/ManaOptions.java

```diff
@@ -1,28 +1,37 @@
 package mage.abilities.mana;
 
+import jdk.nashorn.internal.runtime.regexp.joni.exception.ValueException;
 import mage.ConditionalMana;
 import mage.Mana;
 import mage.abilities.Ability;
+import mage.constants.ManaType;
 import mage.game.Game;
 import mage.game.events.GameEvent;
 import mage.game.events.ManaEvent;
 import mage.game.events.TappedForManaEvent;
 import mage.players.Player;
+import mage.util.TreeNode;
 import org.apache.log4j.Logger;
 
 import java.util.*;
 
+// For .contains problem? https://stackoverflow.com/questions/44062636/hashset-remove-not-working/44062875#44062875
```

Comment:
Is there a more pertinent place to put this comment, or perhaps more context you can provide? It's unclear how this StackOverflow question about HashSet.remove is relevant.